### DISCUSSION TITLE
Client Redistribution After Update

### DISCUSSION THREAD

**Discussion by doxsch** [arkmq-org/arkmq-org-broker-operator#1005] [Q&A] [answered]
We currently have the following setup: two brokers, each with their own PVC running in active-active mode. 

When one broker is updated, the messages are migrated to the remaining broker as expected. Similarly, the client connections are moved to the remaining broker. Subsequently, the second broker is updated, and the clients and messages are moved to the first broker. However, at the end of the update, all clients remain on the first broker and are no longer distributed. How can these clients be redistributed evenly?

**Comment by gtully** [comment, marked as answer]
not easily. 
there are lots of tradeoffs at play and lots of potential variables. For example, if producers and consumers are collocated, there may be no benefit to distributing connections! See the Data Gravity section in the docs: https://activemq.apache.org/components/artemis/documentation/latest/connection-routers.html#data-gravity for some related discussion.

However, a potentially simple solution, is to periodically close the connections on the broker to force the clients to re evaluate. 

There is a recent plugin that will give you that feature.
https://issues.apache.org/jira/browse/ARTEMIS-4709

docs at: https://activemq.apache.org/components/artemis/documentation/latest/broker-plugins.html#using-the-connectionperiodicexpiryplugin

**Reply by doxsch** [reply]
Thank you for your answer. I think we’ll probably use the plug-in.
